### PR TITLE
fitting for xsvs data 

### PR DESCRIPTION
**Pull request by sameera2004**
Negative binomial distribution
Poisson distribution 
Gamma distribution

### CODE REVIEW

File: skxray/core/speckle_fitting.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    """
+    co_eff = np.exp(gammaln(bin_edges + M) -
+                    gammaln(bin_edges + 1) - gammaln(M))
+
+    poisson_gamma = co_eff * np.power(M / (K + M), M) * np.power(K / (M + K),
+                                                                 bin_edges)
+    return poisson_gamma
+
+
+def poisson_distribution(bin_edges, K):
```
Comment (ericdill) [inline skxray/core/speckle_fitting.py]: Why are you not using scipy.stats.poisson?

File: skxray/core/speckle_fitting.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    nbinom_distribution() function Notes
+
+    :math ::
+        P(K) = \frac{<K>^K}{K!}\exp(-K)
+    """
+    poisson_dist = np.exp(-K) * np.power(K, bin_edges)/gamma(bin_edges + 1)
+    return poisson_dist
+
+
+def gamma_distribution(bin_edges, K, M):
```
Comment (ericdill) [inline skxray/core/speckle_fitting.py]: Does this do anything that `scipy.stats.gamma` does not do?

http://docs.scipy.org/doc/scipy-0.15.1/reference/generated/scipy.stats.gamma.html

**Comment by ericdill** [comment]
Don't all of these functions already exist in scipy?

**Comment by licode** [comment]
I suggest to put these functions into lineshapes https://github.com/scikit-xray/scikit-xray/blob/master/skxray/core/fitting/lineshapes.py if they are not included in scipy.

**Comment by danielballan** [comment]
+1 to Li's comment

**Comment by sameera2004** [comment]
@Li yes in the future I will add these fitting functions to lineshapes https://github.com/scikit-xray/scikit-xray/blob/master/skxray/core/fitting/lineshapes.py
@ericdill yes it looks like most of these functions already exist in scipy. Thanks! 
Looks like two are not there, anyway I will check back in scipy if so I will put new PR,

**Pull request closed without merging**